### ISSUE TITLE
Ember 3.17 : error in animateOut

### ISSUE THREAD

**Issue by oliverlj** [cibernox/ember-power-select#1338]
Hello,
When I update to 3.17. i got a parentElement undefined in 
```js
    animateOut(dropdownElement) {
      if (!this.animationEnabled) return;
      let parentElement = this.renderInPlace ? dropdownElement.parentElement.parentElement : dropdownElement.parentElement;
      let clone = dropdownElement.cloneNode(true);
      clone.id = `${clone.id}--clone`;
      clone.classList.remove(...this.transitioningInClass.split(' '));
      clone.classList.add(...this.transitioningOutClass.split(' '));
      parentElement.appendChild(clone);
      this.set('animationClass', this.transitioningInClass);
      waitForAnimations(clone, function () {
        parentElement.removeChild(clone);
      });
    }
```
I use ember-bootstrap-power-select so i don't know yet which one is responsible

**Comment by oliverlj** [comment]
settings animationEnabled to false does nothing as
```js
    // CPs
    get animationEnabled() {
      if (this._animationEnabled) {
        return this._animationEnabled;
      }

      let config = Ember.getOwner(this).resolveRegistration('config:environment');
      return config.environment !== 'test';
    }
```
this return true

**Comment by it-delta** [comment]
Hello!
Yes, I see the problem too...

**Comment by oliverlj** [comment]
no error in the test because animationEnabled is false for test environment

**Comment by sinankeskin** [comment]
Same here with v3.17

**Comment by aonawale** [comment]
Same here ember v3.17

**Comment by rwwagner90** [comment]
Same here with 3.17 as well

**Comment by williamweckl** [comment]
Same here with 3.17.

**Comment by williamweckl** [comment]
> no error in the test because animationEnabled is false for test environment

shouldn't be enabled in tests too? This error was silently deployed in production event with a high test coverage :cry:

**Comment by markusm7** [comment]
Same here with Ember 3.17 and EPS 3.0.6
I see it got tackled in some way in EBD 3.

Any plans to fix this in EPS 3.x.x (or EBD 2.x.x)?

**Comment by leondmello** [comment]
I got rid of these errors by updating to https://github.com/cibernox/ember-power-select/releases/tag/v4.0.0-beta.6

**Comment by markusm7** [comment]
@leondmello yeah as i said i got tackled in ember basic dropdown 3, which is the one EPS 4-beta is using. The problem is im using different addons which rely on EBD 2.

**Comment by leondmello** [comment]
> @leondmello yeah as i said it got tackled in ember basic dropdown 3, which is the one EPS 4-beta is using. The problem is im using different addons which rely on EBD 2.

Yes, I wrote the suggestion for all the other people for whom it might get fixed by updating to the BETA.

**Comment by cibernox** [comment]
I plan to make the beta final today, after I merge a couple PRs. Hopefully that will invite people using ember octane to update

**Comment by markusm7** [comment]
updated all the dependencies today already to ebd 3 and eps 4-beta. glad i moved partially already to octane, seems to continue working like before!

**Comment by cibernox** [comment]
I released 4.0.0 final, so everyone having problems in the very latest version of Ember should be able to update and have them fixed. There is barely any breaking changes, the component uses octane idioms, glimmer components, typescript, @tracked and all that jazz.

**Comment by cibernox** [comment]
I think this can be closed now, this is fixed in 4.0.0
